Create aio-script.sh

codetronaut opened the pull request:
created a script to automate the installation of samba sharing server without login credentials.

codetronaut committed:
Create samba_config.sh

created a script to automate the installation of samba sharing server without login credentials.

amanharitsh123 commented:
@codetronaut 

- Can you please explain why you are downloading an additional config file for samba? 
- Why don't we just append an entry for the shared folder in the predefined samba config file?
- Please have a look at the Readme file, It is mentioned there that the script should work for both fedora and Debian based distros. Your script uses aptitude package manager, kindly make appropriate changes for Fedora as well. 
- Take user input for Folder to be shared and password.

codetronaut committed:
Updated  the whole script

updated the whole samba_config.sh script.

amanharitsh123 commented:
- Please do a test run of the script before committing changes. You have made some blunder mistakes like missing space between "[" and "-z", spaces before and after "=".
- Ask for root permissions beforehand rather than using sudo.
- The script itself will do 4 stated tasks. It should accept options like -1,-2,-3,-4 to perform different tasks. So either program you permission and folder inputs accordingly as arguments or just take input from the user interactively after the script runs.
-  Check if the directory provided by the user actually exists or not.
- eth0 is not guaranteed to be the default ethernet interface. https://unix.stackexchange.com/questions/134483/why-is-my-ethernet-interface-called-enp0s10-instead-of-eth0
- Here is a solution for the same: https://github.com/amanharitsh123/sugarizer-school-box/blob/b9d3646d1f0e5abbe4bf4521aa030b35f3ec669b/image_scripts/hotspot.sh#L9
- Please mention the purpose of line 59 as a comment.

codetronaut committed:
Update samba_config.sh

codetronaut committed:
Create aio-script.sh

created the basic template for all in one script will add functions in next commits.

codetronaut committed:
Delete samba_config.sh

codetronaut reviewed (commented):
Please review  the script.

codetronaut committed:
Update aio-script.sh

amanharitsh123 commented:
Great Job, Options seems to work as expected. Some suggestions to work on further:

- Script throws an error for undefined function usage.
- Raise an error when the script has executed without specifying an option.
- Complete each function in different commit, it will be helpful to find bugs.
- Try to complete functionalities by 5 Feb before the coding period ends.

codetronaut committed:
Update aio-script.sh

Added empty parameter exception.

codetronaut committed:
Update aio-script.sh

Added update and upgrade function.

amanharitsh123 commented on aio-script.sh:
```diff
@@ -24,6 +23,14 @@ function samba_config()
 	echo "Updating..."
+if [ -f /etc/os-release ];
+    then
+        sudo aptitude update && sudo apt upgrade -y
```
Make it apt or apt-get.

codetronaut committed:
Update aio-script.sh

amanharitsh123 commented:
Great. Merging it now.
